From: Gilles Caulier <caulier.gilles@gmail.com>
Date: Sat, 29 Oct 2011 19:04:33 +0000
Subject: update internal libpgf to last stable 6.11.42. Fixbroken compilation due to API changes.
X-Git-Url: http://quickgit.kde.org/?p=digikam.git&amp;a=commitdiff&amp;h=51a92945ea0725fcca9c27435ee852003c441ad0
---
update internal libpgf to last stable 6.11.42. Fixbroken compilation due to API changes.
BUGS: 285240
---


--- a/libs/dimg/loaders/pgfloader.cpp
+++ b/libs/dimg/loaders/pgfloader.cpp
@@ -420,7 +420,12 @@ bool PGFLoader::save(const QString& file
             }
         }
 
+#ifdef PGFCodecVersionID
+#   if PGFCodecVersionID < 0x061142
         header.background.rgbtBlue = header.background.rgbtGreen = header.background.rgbtRed = 0;
+#   endif
+#endif
+
         pgf.SetHeader(header);
 
         pgf.ImportBitmap(4 * imageWidth() * (imageSixteenBit() ? 2 : 1),

--- a/libs/threadimageio/pgfutils.cpp
+++ b/libs/threadimageio/pgfutils.cpp
@@ -131,9 +131,14 @@ bool writePGFImageData(const QImage& img
         header.bpp                  = img.depth();
         header.channels             = 4;
         header.mode                 = ImageModeRGBA;
+
+#ifdef PGFCodecVersionID
+#   if PGFCodecVersionID < 0x061142
         header.background.rgbtBlue  = 0;
         header.background.rgbtGreen = 0;
         header.background.rgbtRed   = 0;
+#   endif
+#endif
         pgfImg.SetHeader(header);
         // NOTE: see B.K.O #273765 : Loading PGF thumbs with OpenMP support through a separated thread do not work properlly with libppgf 6.11.24
         // pgfImg.ConfigureEncoder(false);

